app/src/main/java/com/alphawallet/app/ui/widget/holder/WalletSummaryHolder.java

```diff
@@ -69,19 +90,60 @@ public WalletSummaryHolder(int resId, ViewGroup parent, WalletClickCallback call
         walletAddressText = findViewById(R.id.wallet_address);
         walletClickLayout = findViewById(R.id.wallet_click_layer);
         wallet24hChange = findViewById(R.id.wallet_24h_change);
+        recyclerView = findViewById(R.id.horizontal_list);
+        isMainNetActive = mainNetActivated;
+        tokensServices = tokensService;
         clickCallback = callback;
+        this.context = context;
+        this.tokenRepositoryType = tokenRepositoryType;
+
+
         manageWalletLayout = findViewById(R.id.layout_manage_wallet);
         this.realm = realm;
     }
 
+
+    private void onTokenMetas(Token[] metaTokens)
+    {
+       // tokens.postValue(metaTokens);
+        testNetHorizontalListAdapter = new TestNetHorizontalListAdapter(metaTokens, tokensServices, context);
+        recyclerView.setAdapter(testNetHorizontalListAdapter);
+        testNetHorizontalListAdapter.notifyDataSetChanged();
+    }
+
+    protected void onError(Throwable throwable)
+    {
+        if (throwable instanceof ServiceException)
+        {
+            error.postValue(((ServiceException) throwable).error);
+        }
+        else
+        {
+            String message = throwable.getMessage();
+            if (TextUtils.isEmpty(message))
+            {
+                error.postValue(new ErrorEnvelope(C.ErrorCode.UNKNOWN, null, throwable));
+            }
+            else
+            {
+                error.postValue(new ErrorEnvelope(C.ErrorCode.UNKNOWN, message, throwable));
+            }
+        }
+    }
+
```

Comment:
Please refactor this method; nested if statements should be avoided - you can throw explicit errors and catch them separately.